src/lang/pt/mod.rs

```diff
@@ -0,0 +1,433 @@
+//! Spanish number interpretor
+//! Sources:
+//! - https://www.practiceportuguese.com/
+//! - http://www.portugaisfacile.fr/cours-pour-les-debutants/compter-en-portugais-les-nombres/
+//! - https://www.dicio.com.br/como-escrever-numeros-por-extenso/
+//! - https://exoportugais.blogspot.com/2012/12/nombres-ordinaux-en-portugais.html
+
+use bitflags::bitflags;
+
+use crate::digit_string::DigitString;
+use crate::error::Error;
+
+mod vocabulary;
+
+use super::{LangInterpretor, MorphologicalMarker};
+use vocabulary::INSIGNIFICANT;
+
+#[derive(Default)]
+pub struct Portuguese {}
+
+impl Portuguese {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+bitflags! {
+    /// word chaining restrictions
+    struct Restriction: u64 {
+        const CONJUNCTION = 1;
+        const ONLY_MULTIPLIERS = 2;
+    }
+}
+
+/// pseudo lemmatizer
+fn lemmatize(word: &str) -> &str {
+    if word.ends_with('a') {
+        word.trim_end_matches('a')
+    } else if word.ends_with("as") && word != "duas" {
+        word.trim_end_matches("as")
+    } else if word.ends_with('o') && word != "zero" {
+        word.trim_end_matches('o')
+    } else if word.ends_with("os") {
+        word.trim_end_matches("os")
+    } else {
+        word
+    }
+}
+
+impl LangInterpretor for Portuguese {
```

Comment:
I suppose this is not easy to change now, but we should consider fixing this typo everywhere:
```suggestion
impl LangInterpreter for Portuguese {
```

FYI I learnt about https://github.com/crate-ci/typos while reviewing a Rust PR in an open-source project. This could help with finding typos in our code